src/views/index.js

```diff
@@ -81,13 +85,23 @@ const template = (titlePrefix, ...elements) => {
       )
     );
 
+  const customLink = (filename) => {
+    const customStyleFile = path.join(envPaths("oasis", { suffix: "" }).config, filename);
+    try {
+      if (fs.existsSync(customStyleFile)) {
+        return link({ rel: "stylesheet", href: filename });
```

Comment:
am I reading this would generate a link like this?

```html
<link rel="stylesheet" type="text/css" href="/home/daan/.config/oasis/custom.css">
```

If so, I don't think we can do it like that. Oasis is also run on public instances (most notably probably oasis-demo.fraction.io ) and there this would of course not work, but also try to load a local file into a public website.

I'm pretty sure modern browsers wouldn't allow us to do that anyhow, but it still seems iffy...

Reply:
I'd proposed to add a route to koa that simply fetches `custom.css`, returning an empty document if it doesn't exist.
Kind of like here:

https://github.com/fraction/oasis/blob/master/src/index.js#L617-L627

Specifically, the cache-control can probably be used here too.

Reply:
No the link will reference like:
<link rel="stylesheet" href="/custom-style.css">

And the server will "map" (sorry don't know the correct term) /custom-style.css to /home/daan/.config/oasis/custom-style.css because I added a mapping in src/index.js similar to the way /theme.css is mapped to a file in the @fraction/base16-css module.

/src/index.js line 570 in the commit, where customStyleFile is a const path to the user's file like /home/daan/.config/oasis/custom-style.css:
```js
  .get("/custom-style.css", (ctx) => {
    ctx.type = "text/css";
    ctx.body = requireStyle(customStyleFile);
  })
```

Regarding your reference to loading a blob, I am not entirely sure what is happening with that code, but two things to ask.  Is it returning a 404 on file not found or empty?  That was what I was trying to avoid by checking if the file exists before adding a stylesheet link, so that you don't get a 404 error in the console if the custom-style.css file does not exist.  Not having that file is perfectly acceptable and that error would be confusing and also waste resources for the server trying to get a file which does not need to exist.

Second, am I reading the blob code that is caches the file?  I don't think caching would be a good idea for a css, if I am editing the css file to make tweaks I don't want it to be cached, but see changes I've made immediately on the next refresh.

Sorry this is my first time with node.js code, looks like there are two places that failed checks but do run fine in testing.  I can work on getting those cleaned up.

Reply:
Oh sorry, indeed I skipped those lines with the handler, that looks good then. :)
It will require restarting oasis the first time that the css file is created (to set `haveCustomFile`) but that's fine by me. 

for the cleaning up, I think simply calling `fs.readFileSync(customStyleFile, "utf8");` (without the assignment) should do.